### PR TITLE
feat: add OpenAIChat impl with gpt-4 support

### PR DESCRIPTION
**Pull request by douglas-reid**
Creates a OpenAIChat implementation that can use the new `gpt-4` plugin in Steamship to interact with OpenAI's GPT-4.

### CODE REVIEW

**Comment by EniasCailliau** [comment]
image

This can be fixed by calling .dict() on the Tag's before passing them to json.

**Comment by douglas-reid** [comment]
> image
> 
> This can be fixed by calling .dict() on the Tag's before passing them to json.

This required an upstream fix in python-client.

File: src/steamship_langchain/llms/openai.py [hunk 1]
```diff
@@ -188,3 +189,102 @@ def _batch(
...
+        """Configuration for this pydantic object."""
+
+        extra = Extra.allow
+
+    def __init__(
+        self, client: Steamship, model_name: str = "gpt-4", moderate_output: bool = True, **kwargs
+    ):
+        super().__init__(client=client, model_name=model_name, **kwargs)
+        plugin_config = {"model": self.model_name, "moderate_output": moderate_output}
+        if self.openai_api_key:
```
Comment (eob) [inline src/steamship_langchain/llms/openai.py:207]: Just double checking that Pydantic won't accidentally dump `openai_api_key` to logs if someone calls `print` on it

**Review by eob — APPROVED** [review]
LGTM! Left one safety check comment w/r/t Pydantic & secrets

**Pull request merged**